File: agentepro/backend/app/schemas/metrics.py
from __future__ import annotations

from pydantic import BaseModel


class MessageVolumePoint(BaseModel):
    date: str
    count: int


class LeadsFunnelPoint(BaseModel):
    stage: str
    count: int


class MetricsSummary(BaseModel):
    total_messages: int
    messages_today: int
    messages_change_pct: float
    new_leads: int
    leads_change_pct: float
    appointments_booked: int
    total_calls: int
    avg_response_time_minutes: float
    hot_leads_count: int
    message_volume_chart: list[MessageVolumePoint]
    leads_funnel: list[LeadsFunnelPoint]


class CostBreakdown(BaseModel):
    claude_usd: float
    twilio_usd: float
    retell_usd: float
    fal_usd: float
    total_usd: float
    tokens_used: int
    calls_count: int

File: agentepro/frontend/src/types/metrics.ts
export interface MessageVolumePoint {
  date: string
  count: number
}

export interface LeadsFunnelPoint {
  stage: string
  count: number
}

export interface MetricsSummary {
  total_messages: number
  messages_today: number
  messages_change_pct: number
  new_leads: number
  leads_change_pct: number
  appointments_booked: number
  total_calls: number
  avg_response_time_minutes: number
  hot_leads_count: number
  message_volume_chart: MessageVolumePoint[]
  leads_funnel: LeadsFunnelPoint[]
}
